### PR TITLE
Remove premature abstraction "BufferType"

### PR DESCRIPTION
**Pull request by Kazark**
I would like to implement terminal and scratch buffer support, and
possible others, at some point. At this point, it's just a speculative
construct that is getting in the way by creating currently meaningless
edge cases.

### CODE REVIEW

File: src/Void.Core/BufferList.fs [hunk 1]
```diff
@@ -7,29 +7,23 @@ module Buffer =
         { Filepath = None; Contents = []; CursorPosition = originCell }
 
     let newFile path =
-        BufferType.File { Filepath = Some path; Contents = []; CursorPosition = originCell }
```
Comment (mjgpy3) [inline src/Void.Core/BufferList.fs:10]: Nice. F#, inference is so cool!

File: src/Void.Core/BufferList.fs [hunk 2]
```diff
@@ -7,29 +7,23 @@ module Buffer =
...
-    let empty =
-        BufferType.File emptyFile
-
-    let readLines buffer start =
-        match buffer with
-        | BufferType.File fileBuffer ->
-            fileBuffer.Contents |> Seq.skip (start - 1) // Line numbers start at 1
-        | _ -> Seq.empty
+    let readLines fileBuffer start =
+        fileBuffer.Contents |> Seq.skip (start - 1) // Line numbers start at 1
```
Comment (mjgpy3) [inline src/Void.Core/BufferList.fs:16]: `skip`... Interesting. Other languages tend to say `drop`. I guess `skip` fits better with the vocabulary of a sequence.

**Comment by mjgpy3** [comment]
Is this good to merge?

File: src/Void.Core/BufferList.fs (hunk 2 above)
Comment (Kazark) [inline src/Void.Core/BufferList.fs:16]: Every functional language/library seems to pick a different word. The vocabulary discrepancies for exactly the same things across LINQ/Java 8/Groovy/Scala/Haskell/Underscore.js/F# is a total nightmare.

**Comment by Kazark** [comment]
Yep. If you approve, then merge away!

**Comment by mjgpy3** [comment]
Cool! Do we trust the button or should I wait until I get home?

**Comment by Kazark** [comment]
Eh, you can just trust the button, IMO.

**Pull request merged**